sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferClient.java

```diff
@@ -101,16 +104,105 @@ private TransferClient(Context context,
                         ExistingWorkPolicy.KEEP,
                         uploadWorkRequest)
                     .enqueue();
-                idOrErrorChannel.postValue(TransferIdOrError.id(transferId));
+                idOrErrorChannel.postValue(TransferIdOrError.id(TransferIdOrError.Operation.UPLOAD_DOWNLOAD, transferId));
             } catch (Exception e) {
-                idOrErrorChannel.postValue(TransferIdOrError.error(e));
+                idOrErrorChannel.postValue(TransferIdOrError.error(TransferIdOrError.Operation.UPLOAD_DOWNLOAD, e));
             }
         });
-        return liveDataPair.getTransferInfoLiveData();
+        // UI_Thread
+        return toCachedTransferInfoLiveData(idOrErrorChannel, false);
     }
 
     /**
-     * Get the name for a unique transfer work.
+     * Pause a transfer identified by the given transfer id. The pause operation
+     * is a best-effort and it is possible that a transfer that is already executing
+     * may continue to transfer.
```

Comment:
What causes the behavior described?

Reply:
it's uplifted from WorkManager `cancel` API doc. WorkManager::Worker::cancel() is best effort.